lsp-diagnostics.el

```diff
@@ -298,7 +307,7 @@ See https://github.com/emacs-lsp/lsp-mode."
                       (when (= start end)
                         (if-let ((region (flymake-diag-region (current-buffer)
                                                               (1+ start-line)
-                                                              character)))
+                                                              (1+ character))))
```

Comment:
Can you elaborate why +1? 🤔  Thanks!

Reply:
`flymake-diag-region` expects the line and column values to start at 1, not start at 0 like LSP specifies.  Thus, both the line and column need incrementing when calling `flymake-diag-region`.  The line number was already incremented but the column was not.

Note that this logic is only executed when the LSP server doesn't support an end position (i.e., the start and end position are the same), so `flymake-diag-region` is used to compute it.